### PR TITLE
Code improvements on bullet damage decals

### LINKED ISSUE #36601: Fixing bullet dents in walls is not working

### PR DESCRIPTION
**Pull request by nicbn**
Now fixing the dents is instant too

Maybe fixes #36601

### CODE REVIEW

**Comment by praisenarsie** [comment]
test it

**Comment by nicbn** [comment]
Tested, unable to reproduce the issue, however I'm not sure if this fixes the cause of the issue

File: code/game/turfs/simulated/walls.dm [hunk 1]
```diff
@@ -296,12 +285,22 @@
 	if(LAZYLEN(dent_decals) >= MAX_DENT_DECALS)
 		return
 
-	var/mutable_appearance/decal = pick(dent_decal_list[denttype])
+	var/mutable_appearance/decal = mutable_appearance('icons/effects/effects.dmi', "", BULLET_HOLE_LAYER)
+	switch(denttype)
+		if(WALL_DENT_SHOT)
+			decal.icon_state = "bullet_hole"
+		else
```
Comment (Cyberboss) [inline code/game/turfs/simulated/walls.dm]: should be if(WALL_DENT_HIT)

File: code/game/turfs/simulated/walls.dm [hunk 2]
```diff
@@ -296,12 +285,22 @@
...
+			decal.icon_state = "impact[rand(1, 3)]"
+
 	decal.pixel_x = x
 	decal.pixel_y = y
 
-	cut_overlay(dent_decals)
-	LAZYADD(dent_decals, decal)
+	if(LAZYLEN(dent_decals))
+		cut_overlay(dent_decals)
+		dent_decals.Add(decal)
```
Comment (Cyberboss) [inline code/game/turfs/simulated/walls.dm]: change the .Add to +=

File: code/game/turfs/simulated/walls.dm [hunk 3]
```diff
@@ -198,7 +187,7 @@
 			return FALSE
 
 		to_chat(user, "<span class='notice'>You begin fixing dents on the wall...</span>")
-		if(W.use_tool(src, user, slicing_duration, volume=100))
+		if(W.use_tool(src, user, 0, volume=100))
```
Comment (Cyberboss) [inline code/game/turfs/simulated/walls.dm]: why did you remove slicing_duration

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/turfs/simulated/walls.dm (hunk 3 above)
Comment (nicbn) [reply code/game/turfs/simulated/walls.dm]: It only had it because I forgot to remove when I added this
This is purely visual, having to waste welding fuel is already a fair price for it

File: code/game/turfs/simulated/walls.dm [hunk 4]
```diff
@@ -296,12 +285,22 @@
 	if(LAZYLEN(dent_decals) >= MAX_DENT_DECALS)
 		return
 
-	var/mutable_appearance/decal = pick(dent_decal_list[denttype])
+	var/mutable_appearance/decal = mutable_appearance('icons/effects/effects.dmi', "", BULLET_HOLE_LAYER)
```
Comment (optimumtact) [inline code/game/turfs/simulated/walls.dm]: why are you now constructing it every time instead of picking from the static list?

Comment (nicbn) [reply code/game/turfs/simulated/walls.dm]: I think that reusing the same constructed mutable_appearance was messing up with things. Also the static list thing just looks awful

Comment (optimumtact) [reply code/game/turfs/simulated/walls.dm]: yeah but how does this affect the performance difference?
